lib/srv/desktop/rdp/rdpclient/src/lib.rs

```diff
@@ -120,6 +121,32 @@ pub unsafe extern "C" fn client_run(cgo_handle: CgoHandle, params: CGOConnectPar
     }
 }
 
+/// Gets a [`ClientHandle`] from the global [`CLIENT_HANDLES`] map.
+macro_rules! get_client_handle {
+    ($cgo_handle:expr) => {
+        match get_client_handle($cgo_handle) {
+            Some(it) => it,
+            None => {
+                warn!("call_function_on_handle failed: handle not found");
+                return CGOErrCode::ErrCodeFailure;
+            }
+        }
+    };
+}
+
+/// Handles a function call on a [`ClientHandle`], including logging and error handling.
+macro_rules! handle_operation {
+    ($ctx:expr, $operation:expr) => {
+        match $operation {
+            Ok(_) => CGOErrCode::ErrCodeSuccess,
+            Err(e) => {
+                error!(concat!($ctx, " failed: {:?}"), e);
+                CGOErrCode::ErrCodeFailure
+            }
+        }
+    };
+}
+
```

Comment:
I really don't like macros with flow control (like `return` below), can we replace them both with regular function?
For example:
```suggestion
fn handle_operation<T>(cgo_handle: CgoHandle, ctx: &'static str, f: T) -> CGOErrCode
    where
        T: FnOnce(ClientHandle) -> ClientResult<()>,
{
    let client_handle = match get_client_handle(cgo_handle) {
        Some(it) => it,
        None => {
            warn!("call_function_on_handle failed: handle not found");
            return CGOErrCode::ErrCodeFailure;
        }
    };
    match f(client_handle) {
        Ok(_) => CGOErrCode::ErrCodeSuccess,
        Err(e) => {
            error!("{} failed: {:?}", ctx, e);
            CGOErrCode::ErrCodeFailure
        }
    }
}
```

Reply:
It took a little more finagling to be able to pass in arbitrary arguments:

b4ce750f40e777372a37c5144b4be7a177b5a6bb

Reply:
You don't have to handle variable arguments, compiler can turn closure into `FnOnce`,  with my proposal above call site can look like: 
```rust
Ok(s) => handle_operation(
            cgo_handle,
            "client_update_clipboard",
            |client_handle| client_handle.update_clipboard(s),
        ),
```

Reply:
Oh yeah, my bad.